wallet/src/main/scala/org/bitcoins/wallet/models/AddressTagDAO.scala

```diff
@@ -0,0 +1,71 @@
+package org.bitcoins.wallet.models
+
+import org.bitcoins.core.config.NetworkParameters
+import org.bitcoins.core.crypto.{DoubleSha256Digest, DoubleSha256DigestBE}
+import org.bitcoins.core.protocol.BitcoinAddress
+import org.bitcoins.core.wallet.utxo.AddressTag
+import org.bitcoins.db.{CRUD, SlickUtil}
+import org.bitcoins.wallet.config._
+import slick.jdbc.SQLiteProfile.api._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+abstract class AddressTagDAO[
+    AddressTagType <: AddressTag,
+    DbEntryType <: AddressTagDb[AddressTagType],
+    DbTable <: AddressTagTable[AddressTagType, DbEntryType]](
+    implicit val ec: ExecutionContext,
+    val appConfig: WalletAppConfig)
+    extends CRUD[DbEntryType, BitcoinAddress] {
+  import org.bitcoins.db.DbCommonsColumnMappers._
+
+  val typeName: String
```

Comment:
Why not just define the type here, i.e. 

```
def addressTag: AddressTagType

def typeName: String = addressTag.typeName
```

and then you can get rid of `create(address: BitcoinAddress, tag: AddressTag)`